src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -377,67 +474,106 @@ async def test_TC_TSTAT_4_2(self):
+                logger.info(
+                    "Couldn't run test step 10 since there was no available preset scenario to duplicate")
 
         self.step("11")
         if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
 
-            # Send the AtomicRequest begin command
-            await self.send_atomic_request_begin_command()
+            test_presets = copy.deepcopy(current_presets)
 
-            # Write to the presets attribute after setting the builtIn flag to True for preset with handle (b'\x03')
-            test_presets = copy.deepcopy(new_presets_with_handle)
-            test_presets[2].builtIn = True
+            notBuiltInPresets = list(preset for preset in test_presets if preset.builtIn is False)
 
-            await self.write_presets(endpoint=endpoint, presets=test_presets, expected_status=Status.ConstraintError)
+            if len(notBuiltInPresets) > 0:
+                # Write to the presets attribute after setting the builtIn flag to True for preset with handle (b'\x03')
+                notBuiltInPresets[0].builtIn = True
 
-            # Clear state for next test.
-            await self.send_atomic_request_rollback_command()
+                # Send the AtomicRequest begin command
+                await self.send_atomic_request_begin_command()
+
+                await self.write_presets(endpoint=endpoint, presets=test_presets, expected_status=Status.ConstraintError)
+
+                # Clear state for next test.
+                await self.send_atomic_request_rollback_command()
+            else:
+                logger.info(
+                    "Couldn't run test step 11 since there were no presets that were not built-in")
 
         self.step("12")
         if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
 
-            # Send the AtomicRequest begin command
-            await self.send_atomic_request_begin_command()
+            availableScenarios = list(presetType.presetScenario for presetType in presetTypes if (presetType.presetTypeFeatures & cluster.Bitmaps.PresetTypeFeaturesBitmap.kSupportsNames) != 0 and presetScenarioCounts.get(
+                presetType.presetScenario, 0) < presetType.numberOfPresets)
 
-            # Write to the presets attribute after setting a name for preset with handle (b'\x01') that doesn't support names
-            test_presets = copy.deepcopy(new_presets_with_handle)
-            test_presets[0].name = "Occupied"
+            test_presets = copy.deepcopy(current_presets)
+            presets_without_name_support = list(preset for preset in test_presets if preset.presetScenario in availableScenarios)
 
-            await self.write_presets(endpoint=endpoint, presets=test_presets, expected_status=Status.ConstraintError)
+            if len(presets_without_name_support) == 0 and len(availableScenarios) > 0:
```

Comment:
I'm not sure I'm following this.  `availableScenarios` was the list of scenarios where the preset types _did_ have name support above, no?  So `presets_without_name_support` all have name support...

Am I missing something?  How is this test actually passing?

Reply:
It's passing because it says it has nothing to do, because the clause above was wrong. Fixed.